### PR TITLE
Add ImpureSuite

### PR DESCRIPTION
**Pull request by edmundnoble**
For the impure among us. This would really benefit from #7.

### CODE REVIEW

File: stdlib/src/main/scala/testz/stdlib.scala [hunk 1]
```diff
@@ -74,6 +74,51 @@ object PureSuite {
   }
 }
 
+abstract class ImpureSuite extends Suite {
+  def test[T](test: Test[λ[A => () => Future[A]], () => Future[T]]): T
```
Comment (hrhino) [inline stdlib/src/main/scala/testz/stdlib.scala:78]: what's the point of having the `Future` lifted into a `() => ?`? It's not like it's pure either way.

**Review by hrhino — CHANGES_REQUESTED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Review by hrhino — APPROVED** [review]
> implying there are impure among us

**Comment by hrhino** [comment]
... what the fuck, GitHub

File: stdlib/src/main/scala/testz/stdlib.scala (hunk 1 above)
Comment (edmundnoble) [reply stdlib/src/main/scala/testz/stdlib.scala:78]: a) This is asynchronous and only requires the stdlib.
b) This is as pure as `Task` ;)

**Pull request merged**